### PR TITLE
Improve status command output and improve error handling

### PR DESCRIPTION
**Pull request by hallipr**
Move schema into the tool folder
 
Greatly improve status command output

Fix exception handling in Key Vault stores
- Key Vault 404 exception messages are very long, causing overly verbose console output when a secret is missing

### CODE REVIEW

**Review by weshaggard — APPROVED** [review]
Looks reasonable to me.

File: tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Cli/Commands/StatusCommand.cs [hunk 1]
```diff
@@ -17,13 +20,130 @@ public StatusCommand() : base("status", "Show secret rotation status")
...
+
+        StringBuilder builder = new StringBuilder();
+
+        if (timeSpan.Days > 0)
+        {
+            builder.Append(timeSpan.Days);
+            builder.Append('d');
+        }
+
+        if (timeSpan.Days < 2 && timeSpan.TotalDays - timeSpan.Days > 0)
```
Comment (benbp) [inline tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Cli/Commands/StatusCommand.cs:125]: Do we not print anything if time > 2?

File: tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Tests/TestConfigurations/Valid/random-string.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "$schema": "https://raw.githubusercontent.com/azure/azure-sdk-tools/main/schemas/secretrotation/1.0.0/schema.json",
+  "$schema": "https://raw.githubusercontent.com/azure/azure-sdk-tools/main/tools/secret-rotation/schema/1.0.0/plan.json",
```
Comment (benbp) [inline tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Tests/TestConfigurations/Valid/random-string.json:2]: Is this the new desired pattern for schemas?

File: tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Cli/Commands/StatusCommand.cs (hunk 1 above)
Comment (hallipr) [reply tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Cli/Commands/StatusCommand.cs:125]: We don't add a time component to timeSpan > 2d.

48:01:00 => '2d'
48:00:00 => '2d'
47:59:00 =>  '1d 23h 59m'

File: tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Tests/TestConfigurations/Valid/random-string.json (hunk 2 above)
Comment (hallipr) [reply tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Tests/TestConfigurations/Valid/random-string.json:2]: Moving the schema to the tool folder was requested.

Comment (weshaggard) [reply tools/secret-rotation/Azure.Sdk.Tools.SecretRotation.Tests/TestConfigurations/Valid/random-string.json:2]: Yes I wanted to keep the root of the tools repo clean and keep these scoped under the tool that uses them.

**Pull request merged**